Owncloud iOs Client/Login/Login/WebLoginViewController.swift

```diff
@@ -0,0 +1,202 @@
+            self.retryIfCertificateInCurrentRequestWasAcceptedByUser()
+            return;
+            
+        } else if !self.loadInterrupted {
+            self.error = error
+
+        } //else, let the error set in webView(webView, shouldStartLoadWith, navigationType), if any
+        
+        self.closeLoginViewController()
+    }
+
+    func webView(_ webView: UIWebView, shouldStartLoadWith request: URLRequest, navigationType: UIWebViewNavigationType) -> Bool {
+        
+        let urlToFollow: String = (request.url?.absoluteString)!
+
+        // We store the request to inspect the server certificate and retry it in case of SSL error
+        self.currentRequest = request;
+
+        if urlToFollow.hasPrefix(k_oauth2_redirect_uri){
+            processFinalRedirect(urlToFollow);
+            self.loadInterrupted = true;
+            return false;      // will trigger webView(webView, didFailLoadWithError), with error due to cancellation
+            
+        } else if urlToFollow.hasPrefix(serverPath + k_url_path_list_of_files_in_web) {
+            self.error = UtilsFramework.getErrorByCodeId(Int32(OCErrorOAuth2Error.rawValue))
+            self.loadInterrupted = true;
+            // for some reason, this time "return false" will NOT trigger webView(webView, didFailLoadWithError), with error due to cancellation
+            // so, let's solve it here:
+            self.closeLoginViewController()
+            return false;
+        }
+        
+        return true;
+    }
+    
+    func processFinalRedirect(_ urlToFollow: String) -> Void {
+        if let code = getQueryStringParameter(url: urlToFollow, param: "code") {
+            self.authCode = code
+            self.error = nil;
+            print("contains url and code auth \(self.authCode)")
+            
+        } else if let errorString = getQueryStringParameter(url: urlToFollow, param: "error") {
+            if errorString == "access_denied" {
+                self.error = UtilsFramework.getErrorByCodeId(Int32(OCErrorOAuth2ErrorAccessDenied.rawValue))
+            } else {
+                self.error = UtilsFramework.getErrorByCodeId(Int32(OCErrorOAuth2Error.rawValue))
+            }
+            
+        }
+    }
+    
+    func getQueryStringParameter(url: String, param: String) -> String? {
+        guard let url = URLComponents(string: url) else { return nil }
+        return url.queryItems?.first(where: { $0.name == param })?.value
+    }
+ 
+
+    // MARK - HACK on UIWebView to allow access to servers with insecure certificates that were already accepted by the user
+    
+    func retryIfCertificateInCurrentRequestWasAcceptedByUser() -> Void {
+        // Start an NSURLConnection with the current request to receive an authentication challenge that we can accept or not
+        NSURLConnection.init(request: currentRequest!, delegate: self)
+            // TODO - check if hack works also with NSURLSession
```

Comment:
Another TODO

Reply:
It was not checked yet, let's leave for now